python/fledge/plugins/north/opcuaclient/client.py

```diff
@@ -172,7 +172,16 @@ async def _write_values_to_nodes(self, nodes, values):
         node_identifiers = self._convert_node_identifier(nodes)
         _logger.debug("Nodes: {}".format(node_identifiers))
         _logger.debug("Node Values to write: {}".format(values))
-        asyncio.ensure_future(self.client.write_values(node_identifiers, values))
+
+        try:
+            task = asyncio.create_task(self.client.write_values(node_identifiers, values))
+            # TODO: Timeout interval on the basis of payload block size
+            # Also asyncua do not allow compression
+            await asyncio.wait_for(task, timeout=0.5)
```

Comment:
Better to make this timeout part of configuration; considering large payloads and/or slow network possibilities.

Reply:
Tested with 500 payload block size and it works and even on AWS micro instance. We may consider this for future JIRA' enhancements. Though IMO timeout part should be the part of advanced North configuration instead of plugin configuration.